andebox/actions/changelog_fragment.py

```diff
@@ -0,0 +1,265 @@
+                branch_changes = set(result.stdout.strip().split('\n')) - {''}
+                
+                # Also get staged and unstaged changes
+                result = subprocess.run(
+                    ["git", "diff", "--name-only", "--cached"],
+                    capture_output=True,
+                    text=True,
+                    check=True,
+                )
+                staged_files = set(result.stdout.strip().split('\n')) - {''}
+                
+                result = subprocess.run(
+                    ["git", "diff", "--name-only"],
+                    capture_output=True,
+                    text=True,
+                    check=True,
+                )
+                unstaged_files = set(result.stdout.strip().split('\n')) - {''}
+                
+                # Combine all changes
+                changed_files = branch_changes | staged_files | unstaged_files
+                
+            except subprocess.CalledProcessError:
+                # If comparison fails, fall back to staged and unstaged changes only
+                result = subprocess.run(
+                    ["git", "diff", "--name-only", "--cached"],
+                    capture_output=True,
+                    text=True,
+                    check=True,
+                )
+                staged_files = set(result.stdout.strip().split('\n')) - {''}
+                
+                result = subprocess.run(
+                    ["git", "diff", "--name-only"],
+                    capture_output=True,
+                    text=True,
+                    check=True,
+                )
+                unstaged_files = set(result.stdout.strip().split('\n')) - {''}
+                
+                changed_files = staged_files | unstaged_files
+            
+            return changed_files
+        except subprocess.CalledProcessError as e:
+            raise AndeboxException(f"Failed to get changed files: {e}")
+
+    @staticmethod
+    def is_plugin_file(file_path: str) -> bool:
+        """Check if a file is a plugin file based on its path."""
+        plugin_paths = [
+            "plugins/modules/",
+            "plugins/module_utils/",
+            "plugins/lookup/",
+            "plugins/filter/",
+            "plugins/test/",
+            "plugins/callback/",
+            "plugins/connection/",
+            "plugins/inventory/",
+            "plugins/action/",
+            "plugins/cache/",
+            "plugins/become/",
+            "plugins/cliconf/",
+            "plugins/doc_fragments/",
+            "plugins/httpapi/",
+            "plugins/netconf/",
+            "plugins/shell/",
+            "plugins/strategy/",
+            "plugins/terminal/",
+            "plugins/vars/",
+        ]
+        return any(file_path.startswith(plugin_path) for plugin_path in plugin_paths)
+
+    def get_plugin_changes(self, changed_files: Set[str]) -> List[str]:
+        """Filter changed files to only include plugin files."""
```

Comment:
this method is quite small and requires no comment. remove this (and apply the same logic in the rest of the file)